actions/checkForAndPlaceOrder.ts

```diff
@@ -319,20 +334,82 @@ async function _getTradeableOrderWithSignature(
       owner,
       conditionalOrder.params,
       offchainInput,
       proof
-    )
-    .catch((e) => {
-      console.error(
-        '[getTradeableOrderWithSignature] Error during "getTradeableOrderWithSignature" call: ',
-        e
-      );
-      throw e;
-    });
+    );
+
+    return { result: CallResult.Success, deleteConditionalOrder: false, data };
+  } catch (error: any) {
+    // Print and handle the error
+    // We need to decide if the error is final or not (if a re-attempt might help). If it doesn't, we delete the order
+    const { result, deleteConditionalOrder } = _handleGetTradableOrderCall(
+      error,
+      owner
+    );
+    return {
+      result,
+      deleteConditionalOrder,
+      errorObj: error,
+    };
+  }
+}
+
+function _handleGetTradableOrderCall(
+  error: any,
+  owner: string
+): {
+  result: CallResult.Failed | CallResult.FailedButIsExpected;
+  deleteConditionalOrder: boolean;
+} {
+  if (error.code === Logger.errors.CALL_EXCEPTION) {
+    const errorMessagePrefix =
+      "[getTradeableOrderWithSignature] Call Exception";
+    switch (error.errorName) {
+      case "OrderNotValid":
+        // The conditional order has not expired, or been cancelled, but the order is not valid
+        // For example, with TWAPs, this may be after `span` seconds have passed in the epoch.
+        return {
+          result: CallResult.FailedButIsExpected,
+          deleteConditionalOrder: false,
+        };
+      case "SingleOrderNotAuthed":
+        // If there's no autorization we delete the order
+        // - One reason could be, because the user CANCELLED the order
+        // - for now it doesn't support more advance cases where the order is auth during a pre-interaction
+
+        console.info(
+          `${errorMessagePrefix}: Single order on safe ${owner} not authed. Deleting order...`
+        );
+        return {
+          result: CallResult.FailedButIsExpected,
+          deleteConditionalOrder: true,
+        };
+      case "ProofNotAuthed":
+        // If there's no autorization we delete the order
+        // - One reason could be, because the user CANCELLED the order
+        // - for now it doesn't support more advance cases where the order is auth during a pre-interaction
+
+        console.info(
+          `${errorMessagePrefix}: Proof on safe ${owner} not authed. Deleting order...`
+        );
+        return {
+          result: CallResult.FailedButIsExpected,
+          deleteConditionalOrder: true,
+        };
+    }
+
+    console.error(errorMessagePrefix + " for unexpected reasons", error);
+    // If we don't know the reason, is better to not delete the order
+    return { result: CallResult.Failed, deleteConditionalOrder: false };
+  }
+
+  console.error("[getTradeableOrderWithSignature] Unexpected error", error);
+  // If we don't know the reason, is better to not delete the order
+  return { result: CallResult.Failed, deleteConditionalOrder: false };
```

Comment:
One of the main failure conditions that we may get here is that the `composableCow` contract actually isn't a composableCow contract containing the `getTradeableOrderWithSignature` function. This will lead to `EvmError` issues.

One way to get around this would be in the `addContract` to actually filter on the basis that we *expect* the `getTradeableorderWithSignature` function to exist, so check for the function's selector in `getCode` for the address.

This is a suggestion to help reduce potential issues :smile:

Reply:
100%, we should do this, it is a problem that we show too many errors. 
Ideally this project should only throw when there's sth meaningful to complain about. 

I'll add the issue and probably reiterate soon on that soon.

Reply:
https://github.com/cowprotocol/composable-cow/issues/50